java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -84,35 +84,43 @@ private boolean validate(ValueVector left) {
     return true;
   }
 
-  public boolean visit(BaseFixedWidthVector left) {
+  @Override
```

Comment:
can we add a API here

equals(ValueVector left) { return visit(left, null); }

that way, the callers don't need to pass null.

Reply:
Seems the caller calls ValueVector#accept, visit() is not called explicitly by caller, so add equals is unnecessary?

Reply:
i was trying to simplify the call in DictionalXX files. I'll get back on this.

Reply:
if we add an API
equals(ValueVector left) { return left.accept(this, null); }

we can change this call 
dictionary.accept(new VectorEqualsVisitor(that.dictionary), null);
to
(new VectorEqualsVisitor(that.dictionary)).equals(dictionary)

Reply:
same for all of the usages in TestValueVector and TestRangeValueVector

Reply:
Sounds good to me, And I think it's blocked by https://github.com/apache/arrow/pull/5100 (if we do like this, it will throw UnsupportedException) opened by @liyafan82 , please take a look if we could get that one merged first.

Reply:
I updated this PR, please help take a look, thanks :)

Reply:
@tianchen92 - why do you need the instanceof checks ? This should work, right ?

public boolean equals(ValueVector left) {
    return left.accept(this, null);
 }

Reply:
I guess not, left.accept(this, null) will call accept(ValueVector left, Void value)
1. if we keep accept(ValueVector) it will throw UnsupportedException
2. if we remove accept(ValueVector) it has compatible error
To solve this, we could either check instanceof here or just in accept(ValueVector). What do you think?

Reply:
@pravindra Hi, could you please take a look at this one when available, I have two following up PRs which are based on this one :)  Thanks

Reply:
I tried replacing the fn with 

public boolean equals(ValueVector left) {
  return left.accept(this, null);
}

all of the tests in memory/vector passed. so, I'm missing something ?

Reply:
Sorry I got it wrong, fixed now, thanks!